pkg/apis/logging/v1/output_types.go

```diff
@@ -136,3 +138,23 @@ type Elasticsearch struct {
 	// +optional
 	StructuredTypeName string `json:"structuredTypeName,omitempty"`
 }
+
+// Loki provides optional extra properties for `type: loki`
+type Loki struct {
+	// LabelKeys is a list of meta-data field keys to replace the default Loki labels.
+	//
+	// Loki label names must match the regular expression "[a-zA-Z_:][a-zA-Z0-9_:]*".
+	// Illegal characters in meta-data keys are replaced with "_" to form the label name.
+	// For example the meta-data key "kubernetes.labels.foo" becomes Loki label "kubernetes_labels_foo".
+	//
+	// If LabelKeys is not set, the default labels are:
+	// [log_type, kubernetes_namespace_name, kubernetes_pod_name, kubernetes_host]
+	//
+	// Note: the label kubernetes_host is always included, even if not requested.
+	// Loki requires log streams to be correctly ordered by timestamp.
+	// Including kubernetes_host in the label set ensures each stream originates from a single host,
+	// and timestamps cannot become disordered due to clock differences on different hosts.
+	//
+	// +optional
```

Comment:
nit. Awesome introduction on validation of label keys. If you want to give API docs readers a full insight, I recommend to leave 1 or 2 sentences that the user selection is mandated on loki side by the rate limit config. Precisely these two flags mandate if logs are accepted due to label keys/values length:
```
# Maximum length of a label name.
# CLI flag: -validation.max-length-label-name
[max_label_name_length: <int> | default = 1024]

# Maximum length of a label value.
# CLI flag: -validation.max-length-label-value
[max_label_value_length: <int> | default = 2048]
```
Source: https://grafana.com/docs/loki/latest/configuration/#limits_config